### PR TITLE
Simplify `get-translations.js` and add error handling and fallbacks

### PR DESCRIPTION
**Pull request by dhruvkb**
## Description

This PR improves upon #2188 in the following ways
- breaks the code down into async functions, reducing the use of promises
- breaks the large `get-translations.js` file into smaller files that work independently
- adds error handling at critical points
- adds fallback to separate parallel downloads when bulk download fails
- adds a flag `--require-complete` that can be passed to halt builds when the translations are incompletely downloaded.

All in all, this PR adds significant resilience to the multi-pronged approach developed in #2188.

## Testing Instructions

0. Check out the PR.
1. Try different combinations of the following:
  - Make credentials
    - valid
    - invalid
  - rate limits in `src/locales/scripts/axios.js`
    - reasonable (~50)
    - unreasonable (~1000)
  - `--require-complete` flag
    - set 
    - unset

## Checklist

- [x] My pull request has a descriptive title (not a vague title like
      `Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or
      a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible
      errors.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by zackkrida — CHANGES_REQUESTED** [review]
I'll review more deeply for code, but when I switch locales I'm seeing english, and noticed items in the translation files look like this:

```json
      "filters.audio-categories.sound_effect\u0004Sound effects": {
        "0": "Efectos de sonido"
      },
```

Where the en.json equivalent is:

```json
    "audio-categories": {
      "title": "Audio category",
      "audiobook": "Audiobook",
      "music": "Music",
      "news": "News",
      "podcast": "Podcast",
      "pronunciation": "Pronunciation",
      "sound_effect": "Sound effects",
      "sound": "Sound effects"
    },
```

**Comment by dhruvkb** [comment]
I'll debug that immediately.

File: src/locales/scripts/bulk-download.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    }),
+    {
+      headers: { "content-type": "application/x-www-form-urlencoded" },
+      maxRedirects: 0,
+      validateStatus: () => true,
+    }
+  )
+  if (
+    res.status == 302 &&
+    res.headers["set-cookie"].join(" ").includes("wporg_loggged_in")
```
Comment (zackkrida) [inline src/locales/scripts/bulk-download.js:42]:
```suggestion
    res.headers["set-cookie"].join(" ").includes("wporg_logged_in")
```

**Review by zackkrida — APPROVED** [review]
LGTM. I've tested all cases and it's working well. The logging is all quite clear and useful.

**Pull request merged**